compiler/src/dotty/tools/dotc/transform/Inlining.scala

```diff
@@ -113,7 +100,47 @@ class Inlining extends MacroTransform, IdentityDenotTransformer {
                 else Inlines.inlineCall(tree1)
           else super.transform(tree)
     }
+
+    private def transformMemberDef(tree: MemberDef)(using Context) : Tree =
+      if tree.symbol.is(Inline) then tree
+      else if tree.symbol.is(Param) then
+        super.transform(tree)
+      else if
+        !tree.symbol.isPrimaryConstructor
+        && StagingLevel.level == 0
+        && tree.symbol.hasMacroAnnotation
+      then
+        // Fetch the companion's tree
+        val companionSym =
+          if tree.symbol.is(ModuleClass) then tree.symbol.companionClass
+          else if tree.symbol.is(ModuleVal) then NoSymbol
+          else tree.symbol.companionModule.moduleClass
+
+        // Expand and process MacroAnnotations
+        val (trees, companion) = MacroAnnotations.expandAnnotations(tree, getTracked(companionSym))
+
+        // Enter the new symbols & Update the tracked trees
+        (companion.toList ::: trees).foreach: tree =>
+          MacroAnnotations.enterMissingSymbols(tree, self)
+          updateTracked(tree)
```

Comment:
Do we need this `updateTracked` for the `trees` here? Or is `trees1.foreach(updateTracked)` enough?

I am not sure when we would use the currently tracked trees in `trees.map(super.transform)`.

Reply:
I believe we do, we always have to keep track of the latest available tree. What if `trees` contains a tree that needs the companion. It will also be relevant to MacroAnnotations on parameters